rse-r-package.Rmd

```diff
@@ -0,0 +1,590 @@
+and some functions to merge and plot that data.
+The first step is to run RStudio's project creation wizard,
+select "New Project",
+and then tell RStudio to make it a package (and a Git repository):
+
+FIXME: figure
+
+Before doing our first commit to version control,
+we remove `R/hello.R` and `man/hello.Rd`
+(which the project wizard "helpfully" provides as starting points),
+and add `README.md`, `LICENSE.md`, `CONDUCT.md`, and `CITATION.md`
+to describe the project as a whole,
+its license,
+the contributor code of conduct,
+and how we want the project cited.
+These files are nothing to do with R per se
+(and as we'll see below, R isn't entirely happy having them here with these names),
+but as the [previous book discussed](#FIXME),
+every project should have these files.
+
+We then create a placeholder version of the function we want to write:
+
+```{r, child="rse-r-package/etc/frequency-func.R"}
+```
+
+and then make the following changes to `DESCRIPTION`:
+
+1.  change the `Title`, `Author`, `Maintainer`, and `Description`;
+2.  change the `License` to [`MIT`][mit-license] (see FIXME for other options); and
+3.  go to the `Build` tab in RStudio and run `Check` to see if our package meets CRAN's standards.
+
+```{r, child="rse-r-package/etc/DESCRIPTION-original.md"}
+```
+
+When we run `Check`,
+we get multiple lines of output with one warning and three notes
+in the summary at the bottom:
+
+```
+── R CMD check results ───────────────────────────────────── zipffreq 0.1.0 ────
+Duration: 11.6s
+
+❯ checking for missing documentation entries ... WARNING
+  Undocumented code objects:
+    ‘count_words’
+  All user-level objects in a package should have documentation entries.
+  See chapter ‘Writing R documentation files’ in the ‘Writing R
+  Extensions’ manual.
+
+❯ checking DESCRIPTION meta-information ... NOTE
+  License components which are templates and need '+ file LICENSE':
+    MIT
+
+❯ checking top-level files ... NOTE
+  Non-standard files/directories found at top level:
+    ‘CITATION.md’ ‘CONDUCT.md’ ‘LICENSE.md’
+
+❯ checking package subdirectories ... NOTE
+  Found the following CITATION file in a non-standard place:
+    CITATION.md
+  Most likely ‘inst/CITATION’ should be used instead.
+
+0 errors ✔ | 1 warning ✖ | 3 notes ✖
+Error: R CMD check found WARNINGs
+Execution halted
+
+Exited with status 1.
+```
+
+The explanations for these are:
+
+1.  We haven't written any documentation for the `count_words` function.
+2.  We haven't described our license in the approved way.
+3.  Three of the files we added for GitHub's benefit have to be renamed or moved.
+
+After a bit of searching online,
+we move `LICENSE.md` to `LICENSE` with no extension (but still in the project's root directory)
+and change the entry in `DESCRIPTION` for the license to `License: MIT + file LICENSE` (spelled exactly that way).
+We also add lines to `.Rbuildignore` to tell RStudio to ignore `CITATION.md` and `CONDUCT.md`:
```

Comment:
Use:

```
usethis::use_build_ignore("file-to-ignore")
```
To tell which files to ignore. Its a much more safer approach since `.Rbuildignore` uses pattern finding (so `.` is a regex `.`, not a literal `.`).

Reply:
Fixed.